Library/Homebrew/formula_auditor.rb

```diff
@@ -777,6 +779,21 @@ def audit_specs
       end
     end
 
+    def audit_stable_version
+      return if formula.stable.blank?
+
+      current_version = formula.stable.version
+      current_version_scheme = formula.version_scheme
+
+      collect_previous_info
+
+      if !@newest_committed[:version].nil? &&
```

Comment:
I wonder about having this be:

```suggestion
      if @newest_committed[:version].nil? &&
```

and having `private` `newest_committed` and `previous_whatever` methods which call out to `collect_previous_info` so that:
- don't need to use instance variables all over the place
- don't need to remember to call a method to set them but instead just use the variables and that method will be called if needed

Reply:
We access the hash values quite a bit in the audit methods (four times in `#audit_stable_version` and 19 times in `#audit_revision_and_version_scheme`), so that would add some overhead to accessing those values and I'm not sure it would be worth the tradeoff.

If we simply want to reduce instance variable usage, we could modify `#collect_previous_info` to return a `[@previous_committed, @newest_committed]` array. The audit methods would still need to use `previous_committed, newest_committed = committed_version_info` but that just feels like a cleaner replacement of the existing logic (i.e., calculating those values in a separate method instead of within the audit methods). We would only pay for the cost of fetching the version information once in those two audits, instead of every time we access a value.

In that setup, `#collect_previous_info` would be the only place where the instance variables are used. If it's fine to use the instance variables in `#collect_previous_info` like we're currently doing, then this is a simple change. If we also need/want to minimize use of the instance variables within that method, it would add ~22 lines of code (based on a local refactor).

What do you think about that approach?

Reply:
> so that would add some overhead to accessing those values and I'm not sure it would be worth the tradeoff.

method call overheads are essentially meaningless at this scale. if you mean generating the values: these should be called.

> The audit methods would still need to use `previous_committed, newest_committed = committed_version_info`

This seems better than the status quo and I'd be happy with that 👍🏻